src/sqlfluff/rules/references/RF03.py

```diff
@@ -112,20 +121,17 @@ def _visit_queries(self, query: Query, visited: set) -> Iterator[LintResult]:
                 # :TRICKY: Subqueries in the column list of a SELECT can see tables
                 # in the FROM list of the containing query. Thus, count tables at
                 # the *parent* query level.
-                table_search_root = query.parent if query.parent else query
-                query_list = (
-                    SelectCrawler.get(
-                        table_search_root, table_search_root.selectables[0].selectable
+                possible_ref_tables = list(self._iter_available_targets(query))
+                if query.parent:
+                    possible_ref_tables += list(
+                        self._iter_available_targets(query.parent)
                     )
-                    if table_search_root.selectables
-                    else []
-                )
-                filtered_query_list = [q for q in query_list if isinstance(q, str)]
-                if len(filtered_query_list) != 1:
+                if len(possible_ref_tables) > 1:
                     # If more than one table name is visible, check for and report
```

Comment:
There was no need to run `SelectCrawler.get()` here (which incurs a significant penalty), we can instead just get all the information we need from the `Query` object which has already been parsed.